[Week6][Davidyang2149] Add Restaurant Detail Page - 1차 리팩토링 완료

DavidYang2149 opened the pull request:
# 6강 라우터 - 6주 차 과제 1
## 기능 구현
- [x] TDD로 기능 구현
- [x] 커버리지 100% 달성
- [x] About 페이지 구현
- [x] Not found 페이지 구현
- [x] 레스토랑 상세페이지 구현

## 제한 조건
- [x] 모든 기능은 TDD로 구현합니다. 컴포넌트와 reducer에 대한 테스트가 존재해야 합니다. 먼저 테스트를 추가하고 commit 한 뒤 실제 기능을 추가하고 commit합니다. commit 메시지 로그는 아래와 같이 남아야 합니다.
```sh
Restaurant에 대한 테스트를 추가한다
Restaurant을 구현한다
Restaurant를 리팩토링한다
```
- [x] if는 사용할 수 있어도 else 사용하실 수 없습니다. GuardClauses 방법을 사용해주세요.
- [x] switch는 사용하실 수 없습니다.
- [x] let은 사용하실 수 없습니다. const만을 사용하여 문제를 해결해주세요.
- [x] 함수 이름과 변수 이름에 줄임말은 사용하실 수 없습니다. 길더라도 명확하게 써주세요.
- [x] indent(인덴트, 들여쓰기) depth를 1로 유지해주세요. 예를 들어, for문 안에 if문이 있으면 indent depth는 2입니다. depth를 줄이는 좋은 방법은 함수(또는 메소드)를 분리하면 됩니다.

## 요구사항
- [x] ESLint를 돌린 결과 아무런 문제가 없어야 합니다.
- [x] 모든 인수 테스트를 통과시켜야 합니다.
- [x] 한 파일에는 하나의 컴포넌트만 있어야 합니다.
- [x] 테스트 커버리지 100%를 달성해야 합니다.

## 작업내용 및 질의
- [x] 구현 작업 완료

DavidYang2149 committed:
[Add] Add react-router-dom

6강 라우팅 - 6주 차 과제 1
- package react-router-dom 추가.

DavidYang2149 committed:
[Red] Add App.test.jsx

6강 라우터 - 6주 차 과제 1
- App.test.jsx 헤더 추가 테스트 코드 작성.

DavidYang2149 committed:
[Green] Update App.jsx

6강 라우터 - 6주 차 과제 1
- App.jsx 수정.
- Link 헤더 추가.

DavidYang2149 committed:
[Update] Update index.jsx

6강 라우터 - 6주 차 과제 1
- index.jsx 수정.
- BrowserRouter 추가.

wholemann reviewed (commented):
이제는 별로 어렵지 않으실 거에요ㅎㅎ 쭉쭉 진행해주세요!

DavidYang2149 committed:
[Red] Add RestaurantsPage.test.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantsPage.test.jsx 헤더 추가 테스트 코드 작성.

DavidYang2149 committed:
[Green] Add RestaurantsPage.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantsPage.jsx 추가.
- App.jsx의 컴포넌트 정보 이전.

DavidYang2149 committed:
[Green] Update App.jsx

6강 라우터 - 6주 차 과제 1
- App.jsx 수정.
- RestaurantsPage 추가.

DavidYang2149 committed:
[Red] Update App.test.jsx

6강 라우터 - 6주 차 과제 1
- App.test.jsx 테스트 코드 수정.
- Greend과 Red가 거꾸로 구현됨.

wholemann commented on src/RestaurantsPage.jsx:
```diff
@@ -0,0 +1,35 @@
+
+  return (
+    <div>
+      <header>
+        <Link to="/">
+          <h1>헤더</h1>
+        </Link>
+      </header>
```
아마 이 부분은 따로 빼야 될 겁니다ㅎㅎ

wholemann reviewed (commented):
과제 제출 속도가 너무 더디네요 힘내주세요!!!

DavidYang2149 committed:
[Red] Add HomePage.test.jsx

6강 라우터 - 6주 차 과제 1
- HomePAge.test.jsx 테스트 코드 작성.

DavidYang2149 committed:
[Green] Add HomePage.jsx

6강 라우터 - 6주 차 과제 1
- HomePage.jsx 생성.

DavidYang2149 committed:
[Red] Add AboutPage.test.jsx

6강 라우터 - 6주 차 과제 1
- AboutPage.test.jsx 테스트 코드 작성.

DavidYang2149 committed:
[Green] Add AboutPage.jsx

6강 라우터 - 6주 차 과제 1
- AboutPage.jsx 생성.

DavidYang2149 committed:
[Red] Add NotFoundPage.test.jsx

6강 라우터 - 6주 차 과제 1
- NotFoundPage.test.jsx 테스트 코드 작성.

DavidYang2149 committed:
[Green] Add NotFoundPage.jsx

6강 라우터 - 6주 차 과제 1
- NotFoundPage.jsx 생성.

DavidYang2149 committed:
[Update] Update AboutPage.jsx

6강 라우터 - 6주 차 과제 1
- AboutPage.jsx 수정.

DavidYang2149 committed:
[Red] Update Api.test.jsx

6강 라우터 - 6주 차 과제 1
- Api.test.jsx 테스트 코드 수정.
- 레스토랑 상세 정보 가져오기 테스트 추가.

DavidYang2149 committed:
[Green] Update Api.jsx

6강 라우터 - 6주 차 과제 1
- Api.jsx 수정.
- Restaurant 상세정보 가져오기 추가.

DavidYang2149 committed:
[Red] Update actions.test.jsx

6강 라우터 - 6주 차 과제 1
- actions.test.jsx 테스트 코드 수정.
- 레스토랑 상세 정보 가져오기 테스트 추가.

DavidYang2149 committed:
[Green] Update actions.jsx

6강 라우터 - 6주 차 과제 1
- actions.jsx 수정.
- Restaurant 상세정보 가져오기 추가.

DavidYang2149 committed:
[Update] Update __mocks__/Api.jsx

6강 라우터 - 6주 차 과제 1
- __mocks__/Api.jsx 수정.
- Restaurant 상세정보 가져오기 추가.

DavidYang2149 committed:
[Red] Update reducer.test.jsx

6강 라우터 - 6주 차 과제 1
- reducer.test.jsx 테스트 코드 수정.
- 레스토랑 상세 정보 가져오기 테스트 추가.

DavidYang2149 committed:
[Green] Update reducer.jsx

6강 라우터 - 6주 차 과제 1
- reducer.jsx 수정.
- Restaurant 상세정보 가져오기 추가.

DavidYang2149 committed:
[Red] Update RestaurantsContainer.test.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantsContainer.test.jsx 테스트 코드 수정.
- 레스토랑 상세 정보 가져오기 테스트 추가.

DavidYang2149 committed:
[Green] Update RestaurantsContainer.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantsContainer.jsx 수정.
- Restaurant 상세정보 가져오기 추가.

DavidYang2149 committed:
[Update] Update RestaurantsPage.test.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantsPage.test.jsx 수정.
- Restaurant 상세정보 가져오기 추가.

DavidYang2149 committed:
[Red] Add RestaurantPage.test.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantPage.test.jsx 테스트 코드 생성.

DavidYang2149 committed:
[Green] Add RestaurantPage.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantPage.jsx 생성.

DavidYang2149 committed:
[Red] Add RestaurantContainer.test.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantContainer.test.jsx 테스트 코드 생성.

DavidYang2149 committed:
[Green] Add RestaurantContainer.jsx

6강 라우터 - 6주 차 과제 1
- RestaurantContainer.jsx 생성.

DavidYang2149 committed:
[Red] Update App.test.jsx

6강 라우터 - 6주 차 과제 1
- App.test.jsx 테스트 코드 수정.

DavidYang2149 committed:
[Green] Update App.jsx

6강 라우터 - 6주 차 과제 1
- App.jsx 수정.

DavidYang2149 committed:
[Update] files name change for e2e test

6강 라우터 - 6주 차 과제 1
- e2e 테스트를 위한 라우터명 변경에 따른 파일명 변경.
- 부가적인 설정 조정.

DavidYang2149 commented:
# 진행상황
- e2e 테스트 중  1건 실패.
- 원인 분석 중.

wholemann commented on src/App.jsx:
```diff
@@ -1,32 +1,28 @@
+      <Switch>
+        <Route exact path="/" component={HomePage} />
+        <Route path="/about" component={AboutPage} />
+        <Route path="/restaurants/:restaurantId" component={RestaurantsDetailPage} />
```
```suggestion
        <Route path="/restaurants/:id" component={RestaurantsDetailPage} />
```

REST에서 URL을 표현할 땐 보통 id라고 얘기합니다. 컬렉션인 restaurants에서 식별자(id)로 구분한다는 사실이 중요하죠.

DavidYang2149 replied:
ㅎㅎ 변수명은 항상 독립적으로 생각해야하네요. 감사합니다.

wholemann commented on src/App.test.jsx:
```diff
@@ -4,31 +4,74 @@ import { render } from '@testing-library/react';
 import App from './App';
 
-test('App', () => {
+describe('App', () => {
```
깔끔한 테스트군요ㅎㅎ👍

DavidYang2149 replied:
describe -> it의 절차로 수정하였습니다!

wholemann reviewed (commented):
저희팀에서 같이 일하는 팀원인 줄 알았네요.

이제 저희 스타일에 많이 익숙해지시고 숙련도가 올라오신 것 같습니다.

딱히 얘기드릴 부분이 없어서 뭘 더 얘기드려야 하나 고민했는데

E2E만 모두 통과시키시면 바로 회고 하셔도 될 것 같습니다ㅎㅎ

DavidYang2149 committed:
[Update] Update index.html

6강 라우터 - 6주 차 과제 1
- main.js 절대 경로 수정.

DavidYang2149 commented:
# 구현 완료
- 이번 주는 다행스럽게도 과제를 완료하였습니다!
- 코드리뷰를 토대로 내일 1차 리팩토링을 진행하겠습니다.

DavidYang2149 committed:
[Red] Update actions.test.js

6강 라우터 - 6주 차 과제 1
- Loading 표기를 위한 setLoading 추가.

DavidYang2149 committed:
[Green] Update actions.js

6강 라우터 - 6주 차 과제 1
- Loading 표기를 위한 setLoading 추가.

DavidYang2149 committed:
[Red] Update reducer.test.js

6강 라우터 - 6주 차 과제 1
- Loading 표기를 위한 setLoading 추가.

DavidYang2149 committed:
[Green] Update reducer.js

6강 라우터 - 6주 차 과제 1
- Loading 표기를 위한 setLoading 추가.

DavidYang2149 committed:
[Red] Update RestaurantsDetailContainer.test.js

6강 라우터 - 6주 차 과제 1
- Loading 표기를 위한 setLoading 추가.

DavidYang2149 committed:
[Green] Update RestaurantsDetailContainer.js

6강 라우터 - 6주 차 과제 1
- Loading 표기를 위한 setLoading 추가.

DavidYang2149 committed:
[Update] Update test to describe

6강 라우터 - 6주 차 과제 1
- test를 describe, it으로 변경.

DavidYang2149 committed:
[Refactor] restaurantId -> id change

6강 라우터 - 6주 차 과제 1
- restaurantId 명칭을 id로 변경(URL parameter).

DavidYang2149 commented:
# 1차 리팩토링 완료
- 코드리뷰를 토대로 내일 1차 리팩토링을 진행하였습니다.
- 레스토랑 상세 정보 불러오기에 'Loading'을 추가하였습니다.
- 레스토랑 정보가 없을 경우 '정보가 없습니다.'를 추가하였습니다.

DavidYang2149 commented:
6주 차 회고를 올립니다.

https://davidyang2149.dev/codesoom/%EC%BD%94%EB%93%9C%EC%88%A8-6%EC%A3%BC-%EC%B0%A8-%EC%A3%BC%EA%B0%84%ED%9A%8C%EA%B3%A0/
